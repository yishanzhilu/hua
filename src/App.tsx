/**
 * Copyright (c) 2020 Yishan Authors
 *
 * All rights reserved
 */
import React, { Suspense, lazy } from 'react';
import { Route, Switch } from 'react-router-dom';
import { HelmetProvider } from 'react-helmet-async';

import { BasicError } from 'src/components/errors';

const Home = lazy(() => import('./pages/Home'));
const WorkspaceDashboard = lazy(() => import('./pages/workspace/Dashboard'));

export const HuaApp: React.FC = () => {
  return (
    <HelmetProvider>
      <Suspense fallback={<div>Loading...</div>}>
        <Switch>
          <Route exact path="/" component={Home} />
          <Route path="/workspace/dashboard" component={WorkspaceDashboard} />
          <Route path="*">
            <BasicError statusCode={404} />
          </Route>
        </Switch>
      </Suspense>
    </HelmetProvider>
  );
};
