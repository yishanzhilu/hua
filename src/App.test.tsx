/**
 * Copyright (c) 2019 Yishan Authors
 *
 * All rights reserved
 */
import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

it('App match snapshot', () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});
