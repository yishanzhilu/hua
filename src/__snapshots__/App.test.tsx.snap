// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App match snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="inner"
    >
      <header
        class="header"
      >
        <div
          class="bp3-navbar"
        >
          <div
            class="bp3-navbar-group bp3-align-left"
          >
            <div
              class="bp3-navbar-heading"
            >
              <a
                href="/workspace/dashboard"
              >
                <img
                  alt="logo"
                  src="/logo.svg"
                  style="width: 30px; height: 30px; border-radius: 30px;"
                />
              </a>
            </div>
            <div
              class="bp3-input-group"
            >
              <span
                class="bp3-icon bp3-icon-search"
                icon="search"
              >
                <svg
                  data-icon="search"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    search
                  </desc>
                  <path
                    d="M15.55 13.43l-2.67-2.68a6.94 6.94 0 001.11-3.76c0-3.87-3.13-7-7-7s-7 3.13-7 7 3.13 7 7 7c1.39 0 2.68-.42 3.76-1.11l2.68 2.67a1.498 1.498 0 102.12-2.12zm-8.56-1.44c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
              <input
                class="bp3-input"
                dir="auto"
                placeholder="ÊêúÁ¥¢ÁßªÂ±±"
              />
            </div>
          </div>
          <div
            class="bp3-navbar-group bp3-align-right"
          >
            <a
              class="bp3-button bp3-minimal bp3-intent-primary"
              role="button"
              tabindex="0"
            >
              <span
                class="bp3-button-text"
              >
                ÂÜÖÊµãÁî®Êà∑ÁôªÂΩï
              </span>
            </a>
          </div>
        </div>
      </header>
      <div
        class="body"
      >
        <aside
          class="aside"
        >
          <div
            class="bp3-tree bp3-elevation-0 container"
          >
            <div
              class="container"
            >
              <div
                class="title"
              >
                <div>
                  <div
                    style="border-radius: 3px; width: 20px; height: 20px; background: rgba(55, 53, 47, 0.4); color: rgba(255, 255, 255, 0.9); text-transform: uppercase; display: flex; align-items: center; justify-content: center;"
                  >
                    <div
                      style="line-height: 1; font-size: 14px; height: 14px;"
                    >
                      Q
                    </div>
                  </div>
                </div>
                <div
                  class="bp3-text-overflow-ellipsis"
                  style="font-weight: 600; max-width: 125px;"
                >
                  qy
                </div>
              </div>
              <div
                class="row"
              >
                <span
                  class="bp3-popover-wrapper"
                >
                  <span
                    class="bp3-popover-target"
                  >
                    <span
                      class=""
                      tabindex="0"
                    >
                      Á¥ØËÆ°ÂéÜÁ®ã
                    </span>
                  </span>
                </span>
                <div>
                  <strong>
                    0
                  </strong>
                  <span
                    class="bp3-text-small"
                  >
                    Â∞èÊó∂
                  </span>
                </div>
                <div
                  style="margin-top: 5px;"
                >
                  <div
                    class="bp3-progress-bar bp3-intent-primary bp3-no-animation bp3-no-stripes"
                  >
                    <div
                      class="bp3-progress-meter"
                      style="width: 1%;"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="bp3-divider"
            />
            <nav>
              <div
                class="bp3-tree-node-list"
              >
                <h6
                  class="header"
                >
                  ÂØºËà™
                </h6>
                <a
                  class="bp3-tree-node link"
                  href="/workspace/dashboard"
                >
                  <div
                    class="bp3-tree-node-content"
                    style="padding-right: 20px; padding-left: 20px;"
                  >
                    <span
                      class="bp3-tree-node-icon"
                    >
                      üìã
                    </span>
                    <span
                      class="bp3-text-overflow-ellipsis"
                    >
                      ÁúãÊùø
                    </span>
                  </div>
                </a>
                <a
                  class="bp3-tree-node link"
                  href="/workspace/plan"
                >
                  <div
                    class="bp3-tree-node-content"
                    style="padding-right: 20px; padding-left: 20px;"
                  >
                    <span
                      class="bp3-tree-node-icon"
                    >
                      üóì
                    </span>
                    <span
                      class="bp3-text-overflow-ellipsis"
                    >
                      ËßÑÂàí
                    </span>
                  </div>
                </a>
                <a
                  class="bp3-tree-node link"
                  href="/workspace/success"
                >
                  <div
                    class="bp3-tree-node-content"
                    style="padding-right: 20px; padding-left: 20px;"
                  >
                    <span
                      class="bp3-tree-node-icon"
                    >
                      üèÜ
                    </span>
                    <span
                      class="bp3-text-overflow-ellipsis"
                    >
                      ÊàêÂ∞±
                    </span>
                  </div>
                </a>
                <a
                  class="bp3-tree-node link"
                  href="/workspace/recycle"
                >
                  <div
                    class="bp3-tree-node-content"
                    style="padding-right: 20px; padding-left: 20px;"
                  >
                    <span
                      class="bp3-tree-node-icon"
                    >
                      ‚ôªÔ∏è
                    </span>
                    <span
                      class="bp3-text-overflow-ellipsis"
                    >
                      ÂõûÊî∂Á´ô
                    </span>
                  </div>
                </a>
              </div>
            </nav>
          </div>
        </aside>
        <main
          class="main"
        >
          <div
            class="mainBody"
          >
            <div>
              Loading...
            </div>
          </div>
          <div
            class="mainFooter"
          >
            ¬© 
            2020
             Yishan Authors
          </div>
        </main>
      </div>
    </div>
  </div>
</div>
`;
